### PR TITLE
Improve sessions.sql

### PR DESCRIPTION
**Pull request by inoas**
* more strict types
  - table => utf-8 charset
  - primary-key => ascii_bin
  - data => blob
  - expires => unsigned & could in theory be set to NULL (does not expire)
* added created and modified which are set automatically without having to create the table object (but you can and the DatabaseSession handler will take a `SessionsTable.php` if it can find it). These fields help with server side invalidation.

Ref: https://github.com/cakephp/app/issues/535

### CODE REVIEW

**Comment by ravage84** [comment]
Was this SQL file MySQL specific before? Now it is due to the use of MySQL specific features, such as InnoDB...

**Comment by chinpei215** [comment]
Since there was `int(11)`, it wouldn't support other databases originally.

**Comment by ravage84** [comment]
@chinpei215 Thanks.

Isn't this file what is shown [here](https://book.cakephp.org/3.0/en/development/sessions.html#database-sessions) too?

**Comment by chinpei215** [comment]
Since it says,
> This configuration will require a database table to be added with **at least these fields**:

we would have to change the above sentence if we want to change the scheme declaration in the page.

**Comment by lorenzo** [comment]
We should maybe have a migration instead that can be copied into your own migrations folder

File: config/schema/sessions.sql [hunk 1]
```diff
@@ -5,9 +5,11 @@
...
-  data text,
-  expires INT(11) NOT NULL,
-  PRIMARY KEY  (id)
-);
+CREATE TABLE `sessions` (
+  `id` char(40) CHARACTER SET ascii COLLATE ascii_bin NOT NULL,
+  `created` datetime DEFAULT CURRENT_TIMESTAMP,
+  `modified` datetime DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  `data` blob DEFAULT NULL,
+  `expires` int(11) unsigned DEFAULT NULL,
```
Comment (inoas) [inline config/schema/sessions.sql]: The digits limitation of the integer types in SQL are just for zero-filling, if that exists. However I will still change it, as it won't hurt and the current max unix timestamp is 32bit int (https://en.wikipedia.org/wiki/Unix_time)

**Comment by ravage84** [comment]
I agree with @lorenzo . Wanted to suggest that, anyway.

**Comment by inoas** [comment]
@lorenzo I don't think we should remove the easy way to create the sessions table. However I would welcome a migration-way to do it.

**Comment by lorenzo** [comment]
No, I'm not saying to remove this, just add another way.

**Pull request merged**